x-pack/legacy/plugins/maps/public/connected_components/map/features_tooltip/feature_geometry_filter_form.js

```diff
@@ -61,6 +67,18 @@ export class FeatureGeometryFilterForm extends Component {
       geoFieldType,
       relation,
     });
+
+    // Ensure filter will not overflow URL. Filters that contain geometry can be extremely large.
+    // No elasticsearch support for pre-indexed shapes and geo_point spatial queries.
+    if (window.location.href.length + rison.encode(filter).length > urlOverflow.failLength()) {
+      this.setState({
+        errorMsg: i18n.translate('xpack.maps.tooltip.geometryFilterForm.filterTooLargeMessage', {
+          defaultMessage: 'Unable to create filter, too many vertices in shape.'
```

Comment:
Does it make sense to split this up in two code-paths? 

(1) One for pre-indexed shapes, and one for (2) raw shapes? 

For (2), we can preserve this check with the error message (and even include some wording that you could index the shape first).

For (1), we do _not_ do the check since we assume it's just the default working of Kibana then.

Reply:
I think it will make more sense to have 2 paths once geo_point supports pre-indexed shapes. Right now, the logic is leaky and I would have to check for preindexed shape and that geometry index type is geo_shape. I figured we could add the path when geo_point supportes pre-indexed shaped queries and the check is as simple as seeing if preindexed shape exists.